blog/_posts/2016-10-24-scalafix.md

```diff
@@ -0,0 +1,215 @@
+
+In scalafix, we use the scala.meta API to parse Scala source files,
+traverse abstract syntax trees, inspect tokens (including comments!) and
+then perform the minimal required transformation to the source file.
+Thanks to scala.meta dialects, which abstract away syntactic differences
+between variants of Scala, we can use the same programs to manipulate
+regular Scala source files, SBT files as well as code that uses new
+Dotty syntax such as union types and trait parameters. All this powerful
+functionality is implemented behind a minimalistic interface.
+The NonVolatileLazyVal
+[implementation](https://github.com/scalacenter/scalafix/blob/master/core/src/main/scala/scalafix/rewrite/VolatileLazyVal.scala)
+is 15 lines of code. It could comfortably fit in three tweets.
+
+We would not be able to achieve this level of expressiveness with
+scala.reflect, the current standard metaprogramming framework for Scala.
+Why? To name a few reasons, scala.reflect does not preserve syntactic
+details such as comments and scala.reflect has no notions of dialects,
+so it is bound to only support Scala 2.x.
+
+How does scalafix evolve?
+-----------------------------
+
+As we have seen above, the functionality of scalafix heavily relies on
+the infrastructure provided by scala.meta. However, in order to write
+more sophisticated scalafix rewrite rules, there are two main features
+missing in scala.meta, namely a *scalac converter* and *semantic API*.
+We are closely collaborating with Eugene Burmako, the lead developer of
+scala.meta, in the development efforts of these two features.
+
+### Scalac converter
+
+The scalac converter is a key subsystem of scala.meta. For every Scala
+feature, the converter recognizes its representation in the Scala
+compiler and translates it into the corresponding data structures of the
+scala.meta API.
+
+Over the last months, we’ve made great improvements to the converter. We
+have established a comprehensive test suite that consists of a sample of
+over 26.000 Scala source files collected from popular open-source
+projects. Thanks to the joint effort of the team of scala.meta
+[contributors](https://github.com/scalameta/paradise/graphs/contributors),
+the converter now supports about 19.000 source files form the test
+corpus. We are continuously working to increase that number, aiming to
+bring language coverage to 100%.
+
+### Semantic API
+
+The semantic API will enable scala.meta programs to inspect compiler information
+such as type inference, name resolution and other functionality that
+requires a compilation step. This opens opportunities for new scalafix rewrite
+rules that cannot be done on a purely syntactic level.
+
+We already have a
+[working prototype](https://github.com/scalacenter/scalafix/pull/14) 
+of the first scalafix rewrite that uses the semantic API: ExplicitImplicit.
+ExplicitImplicit inserts type annotations for implicit definitions, like this:
```

Comment:
Again, maybe state that this needs more than syntactic rewrites and explain why. I know this is implicit, but people who don't understand how this all works probably won't make this connection by themselves.